CommuniTEAM/CommuniTEA
# [TEA-11] Implementation of Pre-Commit

### In this PR, I'm adding/changing the following:

- Added a [pre-commit](https://pre-commit.com) configuration file with external hooks:
    - [Commitlint](https://commitlint.js.org/#/)
    - ESLint
- Configured the commit message ruleset to require standardization and Jira issue references
- Configured the ESLint TypeScript ruleset for the frontend, utilizing the [Airbnb config](https://github.com/airbnb/javascript)
- Added default formatting for TypeScript and JavaScript using Prettier in VSCode
- Documentation for Git hook installation and usage added to the project README

### Please do the following to test my code:

1. Delete and rebuild your docker containers to get the new dependencies installed
2. Install pre-commit on your local machine according to [its instructions](https://pre-commit.com/#install)
3. Navigate to the project's root directory in your terminal and run `pre-commit install`
4. Install and enable the VSCode [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
5. Make some changes to any .tsx file in the app directory (try adding a semicolon somewhere!)
6. Run a manual save by hitting Ctrl+S on Windows/Linux or Cmd+S on Mac
7. Prettier should have formatted the file for you! If there were any semicolons, they should have been removed here
8. In the terminal, run `pre-commit run --all-files` to ensure the hooks run
9. Attempt to commit your changes: pre-commit should run the commit hook in addition to the others
10. If you fail the commit lint check, reference the [updated README](https://github.com/CommuniTEAM/CommuniTEA/blob/TEA-11-Implementation-of-Husky/README.md#commit-message-format) on how to format your commits

### Notes:
Please make sure you install and enable VSCode's [ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) so you can see the linting errors in real time -- Airbnb is strict!!
